src/components-v2/vault-detail/actions/MobileStickyActionButtons.tsx

```diff
@@ -29,35 +34,50 @@ interface Props {
 
 export const MobileStickyActionButtons = observer(({ vault, onDepositClick, onWithdrawClick }: Props): JSX.Element => {
   const classes = useStyles();
-  const { vaults, wallet } = React.useContext(StoreContext);
+  const { vaults, wallet, user, chain: networkStore } = React.useContext(StoreContext);
   const canUserDeposit = wallet.isConnected ? vaults.canUserDeposit(vault) : false;
   const canUserWithdraw = vaults.canUserWithdraw(vault);
+  const { depositBalance } = useVaultInformation(vault);
+  const isUserHasToken = user.getBalance(vault.underlyingToken).hasBalance();
+  const isUserHasDeposit = !depositBalance.tokenBalance.eq(0);
+  const { network } = networkStore;
+  const strategy = Chain.getChain(network).strategies[vault.vaultToken];
 
   return (
     <div className={classes.root}>
       <Grid container spacing={1}>
         <Grid item xs>
-          <VaultActionButton
-            fullWidth
-            color="primary"
-            variant={canUserDeposit ? 'contained' : 'outlined'}
-            disabled={!canUserDeposit}
-            onClick={onDepositClick}
-          >
-            Deposit
-          </VaultActionButton>
-        </Grid>
-        <Grid item xs>
-          <VaultActionButton
-            color="primary"
-            variant="outlined"
-            fullWidth
-            disabled={!canUserWithdraw}
-            onClick={onWithdrawClick}
-          >
-            Withdraw
-          </VaultActionButton>
+          {isUserHasToken ? (
+            <VaultActionButton
+              fullWidth
+              color="primary"
+              variant={canUserDeposit ? 'contained' : 'outlined'}
+              disabled={!canUserDeposit}
+              onClick={onDepositClick}
+            >
+              Deposit
+            </VaultActionButton>
+          ) : (
+            <Link href={strategy.depositLink} target="_blank" className={classes.goToLink} underline="none">
```

Comment:
is there any check to verify we have deposit links or what happens if its missing?

Reply:
@axejintao need your assistance here, What to do if no deposit link is available? I can see a few vaults ( wbtc, cvxCRV, graviAURA and auraBAL) for which there are no deposit links.

Reply:
so for ones that are naked deposits there is no link needed - however, for cvxcrv and aurabal we should probably actually link to the minting location (which unfortunately conflicts a bit with the ask i've posted above - however i view these two specifically more as edge cases vs. a normal vault as they are not really LP tokens as much as locked gov tokens) 

i won't get into beVAL vs. aura bal because it IS an LP but i think to link to aura minting page is sufficient because it walks you through the LP addition portion as well 

aurabal (front page)
- https://app.aura.finance/

convex (stake page)
- https://www.convexfinance.com/stake

Reply:
@axejintao In worst cases when there is no deposit link. What should we show?

Reply:
if there is no deposit link we can do 

- show nothing
- link to amm to purchase deposit token (cvx, aura)

i do not think many should not have it - any LP deposit tokens **should** have it added and any single deposits will have it missing although off the top of my head we should only have those two atm i think

Reply:
I think we should always have a button there so users know where it is, can disable or not but it should be there as at least a default.